src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -69,28 +68,37 @@ internal void InitializeLicense(string licenseText)
             license = foundLicense;
         }
 
-        static bool FirstTimeUser()
+        static bool NeedsReporting()
         {
             // Check for the presence of HKCU\Software\NServiceBus
-            using (var regNsbRoot = Registry.CurrentUser.OpenSubKey(@"Software\NServiceBus"))
+            using (var nsbRootRegKey = Registry.CurrentUser.OpenSubKey(@"Software\NServiceBus"))
             {
-                if (regNsbRoot == null)
+                if (nsbRootRegKey != null) return false;
+            }
+
+            //Check for the presence of HKCU\Software\ParticularSoftware
+            using (var particularRegKey = Registry.CurrentUser.OpenSubKey(@"Software\ParticularSoftware"))
+            {
+                // Check for the presence of HKCU\Software\ParticularSoftware\PlatformInstaller
+                using (var platformInstallerRegKey = particularRegKey?.OpenSubKey(@"Software\ParticularSoftware"))
                 {
-                    //Check for the presence of HKCU\Software\ParticularSoftware
-                    using (var regParticularRoot = Registry.CurrentUser.OpenSubKey(@"Software\ParticularSoftware"))
-                    {
-                        if (regParticularRoot == null)
-                        {
-                            return true;
-                        }
-                    }
+                    if (platformInstallerRegKey != null) return false;
                 }
+
+                // Check if NuGetUser value is set. Previous versions of NServiceBus Nuget installs creates this value.
+                var isNsbPreviouslyInstalled = particularRegKey?.GetValue("NuGetUser");
+                if (isNsbPreviouslyInstalled != null) return false;
             }
-            return false;
+            return true;
         }
 
         static async Task TrackFirstTimeUsageEvent()
         {
+            // Set the regisry key for NuGetUser and then call the web api. Web api does not need to succeed.
+            // We only attempt once. Future executions will check the presence of this value to 
+            // ensure that we don't call the web api more than once.
+            SetupFirstTimeRegistryKeys();
```

Comment:
maybe I didn't see it, but it looks like we're not catching any exceptions which might occur by the registry access. Is this something we want to even prevent startup of the endpoint or should that be swallowed similar to a failing web request?